src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -2447,6 +2447,10 @@ private void seekPastOversizedValue() {
 
     @Override
     public Event nextValue() {
+        if (event != Event.NEEDS_DATA) {
+            annotationSequenceMarker.startIndex = -1;
+            annotationSequenceMarker.endIndex = -1;
+        }
```

Comment:
Does it work to put this in `reset()`?

Reply:
Can you please add a 1.0 test for this too and try it on the master branch? If the problem exists there I want to fix it before we eventually merge in the ion-11-encoder branch.

Reply:
It does work in `reset()`. I am adding a test for 1.0 on this branch—without the fix it is not failing for Ion 1.0.
I actually only stumbled across this bug because it's covered by `ion-tests`, so I'm pretty confident that it's fine on `master`, but I can double check that.